util/file.py

```diff
@@ -224,18 +225,25 @@ def mkdir_p(dirpath):
             raise
 
 
-def open_or_gzopen(fname, *opts):
-    return fname.endswith('.gz') and gzip.open(fname, *opts) or open(fname, *opts)
+def open_or_gzopen(fname, *opts, **kwopts):
+    # 'U' mode is deprecated in py3 and may be unsupported in future versions, 
+    # so use newline=None when 'U' is specified
+    for opt in opts:
+        if type(opt) == str and 'U' in opt and sys.version_info[0] == 3:
+            if 'newline' not in kwopts:
+                kwopts['newline'] = None
+            break
+    return fname.endswith('.gz') and gzip.open(fname, *opts, **kwopts) or open(fname, *opts, **kwopts)
 
 
 def read_tabfile_dict(inFile):
     ''' Read a tab text file (possibly gzipped) and return contents as an
         iterator of dicts.
     '''
-    with open_or_gzopen(inFile, 'rt') as inf:
+    with open_or_gzopen(inFile, 'rU') as inf:
         header = None
         for line in inf:
-            row = [item.strip() for item in line.rstrip('\n').split('\t')]
+            row = [item.strip() for item in line.rstrip('\n').rstrip('\r').split('\t')]
```

Comment:
this could just be `line.rstrip('\n\r').split('\t')` to reduce function calls (`rstrip` considers its argument to be an unordered list of characters to strip). also below in `read_tabfile`